src/modules/camera_feedback/CameraFeedback.cpp

```diff
@@ -112,11 +115,37 @@ CameraFeedback::Run()
 		}
 
 		// Fill attitude data
-		// TODO : this needs to be rotated by camera orientation or set to gimbal orientation when available
-		capture.q[0] = att.q[0];
-		capture.q[1] = att.q[1];
-		capture.q[2] = att.q[2];
-		capture.q[3] = att.q[3];
+		if (gimbal.timestamp > 0) {
```

Comment:
Require that gimbal isn't stale? 

```suggestion
		if (_gimbal_sub.copy(&gimbal) && (hrt_elapsed_time(&gimbal.timestamp) < 1_s)) {
```

Reply:
^maybe with a `static constexpr hrt_abstime kGimbalTimeout = 1_s` to avoid magic numbers :)

Reply:
The point of avoiding magic numbers is to capture some kind of explanation in the naming of the constant and to avoid duplication or inconsistencies (among other things).

I would argue in simple one off cases moving to a named constant (that's not defined near the actual usage) can end up being worse and hurts readability/comprehensibility. 

(I don't actually care in this case, just something to think about in general)

Reply:
`gimbal_device_attitude_status` is published by the MAVLink receiver when the gimbal v2 protocol is used and the timestamp is set to the time since boot of the sender. https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/mavlink_receiver.cpp#L3131
Any concerns if I change it to `gimbal_attitude_status.timestamp = hrt_absolute_time()`? The message is forwarded via MAVLink which would preserve the timestamp for other receivers and in PX4 other gimbal related messages set it already to `hrt_absolute_timestamp()` instead of `time_boot_ms` (e.g. https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/mavlink_receiver.cpp#L3092) so it would be more consistent in the log.

Reply:
> I would argue in simple one off cases moving to a named constant (that's not defined near the actual usage) can end up being worse and hurts readability/comprehensibility.

You're right. But I always just default to these practices, as one offs quickly become two- and three- offs in PX4 code blocks. And if the constants are defined in the header at the top, it's generally not hard to find them (also using a nice IDE one can just right click or hover to see the number).

For this case really isn't a big deal though. just my standard pedantic pickiness.